sdk/core/azure-core/inc/http/curl/curl.hpp

```diff
@@ -48,14 +117,32 @@ namespace Azure { namespace Core { namespace Http {
       // insert headers
       for (auto header : headers)
       {
-        // TODO: check result is not null or trow
         headerList = curl_slist_append(headerList, (header.first + ":" + header.second).c_str());
+        if (headerList == NULL)
+        {
+          throw;
```

Comment:
It does not seem to be inside a catch block, but you are using rethrow. You need to throw something.

Reply:
weak request